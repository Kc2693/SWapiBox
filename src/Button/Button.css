
#category-button {
  color: var(--button-text-color);
  background-color: var(--main-light-red);
  border-radius: 5px;
  border-color: var(--main-yellow);
  font-size: 20px;
  padding: 5px 10px;
  text-decoration: none;
  outline: none;
  z-index: 20;
}

#category-button:hover {
  color: var(--main-yellow);
  background-color: var(--main-dark-red);
}

#faves {
  font-size: 24px;
  background-color: var(--main-light-red);
  border-radius: 5px;
  color: var(--button-text-color);
  padding: 5px 10px;
  position: absolute;
  text-decoration: none;
  right: 10px;
  top: 10px;
}
.selected {
  text-decoration: none;
  color: #ff0066;
}
