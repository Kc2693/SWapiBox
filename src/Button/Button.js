import React, { Component } from 'react';
import './Button.css';

class Button extends Component {

  buttonClicker = (event) => {
    this.props.controlFunc(event.target.name)
    this.props.secondFunc()
  }

  render () {
    return (
      <div>
        <button name={this.props.name}
                id={this.props.id}
                onClick={this.buttonClicker}>{this.props.name}{this.props.faveNum}</button>
      </div>
    )
  }
}



export default Button;
