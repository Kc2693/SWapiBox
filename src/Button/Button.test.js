import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import { shallow } from 'enzyme';

describe('Button', () => {
  let wrapper;
  let props;
  let mockEvent;
  let controlFunc = jest.fn();
  let fetchFunc = jest.fn();

  beforeEach(() => {
    props = {"controlFunc": controlFunc,
             "fetchFunc": fetchFunc,
             "id": "category-button",
             "name": "people"
           }
    wrapper = shallow(<Button {...props}/>);
  });


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('buttonClicker should run controlFunc and fetchFunc with params ', () => {
    mockEvent = {preventDefault: jest.fn(), target: {"name": "people"}}

    wrapper.instance().buttonClicker(mockEvent);

    expect(controlFunc).toHaveBeenCalledWith(mockEvent.target.name);
    expect(fetchFunc).toHaveBeenCalledWith(mockEvent.target.name);
  });
});
