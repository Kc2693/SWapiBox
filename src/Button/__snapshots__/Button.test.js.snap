// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    controlFunc={[Function]}
    fetchFunc={[Function]}
    id="category-button"
    name="people"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        id="category-button"
        name="people"
        onClick={[Function]}
>
        people
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "people",
          undefined,
        ],
        "id": "category-button",
        "name": "people",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "people",
        undefined,
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          id="category-button"
          name="people"
          onClick={[Function]}
>
          people
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "people",
            undefined,
          ],
          "id": "category-button",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "people",
          undefined,
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
